.github/workflows/maven.yml

```diff
@@ -0,0 +1,25 @@
+name: Thunder CI pipeline
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK 21
+        uses: actions/setup-java@v3
+        with:
+          java-version: '21'
+          distribution: 'temurin'
+          cache: maven
+      - name: Clean and Compile
+        run: mvn -B clean compile --file pom.xml
```

Comment:
We can skip the call of the 'compile' phase since 'mvn test' automatically goes through all preceding lifecycle phases, including 'compile'. Calling 'mvn test' should handle both compilation and testing.